extensions/functions_aggregate_approx.yaml

```diff
@@ -0,0 +1,17 @@
+%YAML 1.2
+---
+aggregate_functions:
+  - name: "approx_count_distinct"
+    description:  >-
+      Calculates the approximate number of rows that contain distinct values of the expression argument using
+      HyperLogLog. This function provides an alternative to the COUNT (DISTINCT expression) function, which
+      returns the exact number of rows that contain distinct values of an expression. APPROX_COUNT_DISTINCT
+      processes large amounts of data significantly faster than COUNT, with negligible deviation from the exact
+      result.
+    impls:
+      - args:
+          - value: any
+        nullability: DECLARED_OUTPUT
+        decomposable: MANY
+        intermediate: i64
```

Comment:
I don't think the intermediate will be i64. I suspect it will be a binary value.

Reply:
Also, can you share example systems that have this. We should all agree that this is common enough that we include in Substrait versus via independent extension. Thanks!

Reply:
Good catch. I've updated that